test/e2e/cluster_upgrade.go

```diff
@@ -170,191 +257,77 @@ func nodeUpgradeGKE(v string) error {
-			By("Performing a node upgrade")
-			// Circumnavigate testUpgrade, since services don't necessarily stay up.
-			framework.Logf("Starting upgrade")
-			framework.ExpectNoError(nodeUpgrade(f, replicas, v))
-			framework.Logf("Upgrade complete")
-			By("Checking node versions")
-			framework.ExpectNoError(checkNodesVersions(f.Client, v))
-			By("Validating cluster after node upgrade")
-			framework.ExpectNoError(validate(f, svcName, rcName, ingress, replicas))
-		})
+func testServiceRemainsUp(f *framework.Framework, sem *chaosmonkey.Semaphore) {
+	testService(f, sem, true)
+}
 
-		It("should maintain responsive services [Feature:ExperimentalNodeUpgrade]", func() {
-			By("Validating cluster before node upgrade")
-			framework.ExpectNoError(validate(f, svcName, rcName, ingress, replicas))
-			By("Performing a node upgrade")
-			testUpgrade(ip, v, func(v string) error {
-				return nodeUpgrade(f, replicas, v)
-			})
-			By("Checking node versions")
-			framework.ExpectNoError(checkNodesVersions(f.Client, v))
-			By("Validating cluster after node upgrade")
-			framework.ExpectNoError(validate(f, svcName, rcName, ingress, replicas))
-		})
+// testService is a helper for testServiceUpBeforeAndAfter and testServiceRemainsUp with a flag for testDuringUpgrade
+//
+// TODO(ihmccreery) remove this abstraction once testServiceUpBeforeAndAfter is no longer needed, because node upgrades
+// maintain a responsive service.
+func testService(f *framework.Framework, sem *chaosmonkey.Semaphore, testDuringUpgrade bool) {
+	// Setup
+	serviceName := "service-test"
+
+	jig := NewServiceTestJig(f.Client, serviceName)
+	// nodeIP := pickNodeIP(jig.Client) // for later
+
+	By("creating a TCP service " + serviceName + " with type=LoadBalancer in namespace " + f.Namespace.Name)
+	// TODO it's weird that we have to do this and then wait WaitForLoadBalancer which changes
+	// tcpService.
+	tcpService := jig.CreateTCPServiceOrFail(f.Namespace.Name, func(s *api.Service) {
+		s.Spec.Type = api.ServiceTypeLoadBalancer
 	})
+	tcpService = jig.WaitForLoadBalancerOrFail(f.Namespace.Name, tcpService.Name)
+	jig.SanityCheckService(tcpService, api.ServiceTypeLoadBalancer)
+
+	// Get info to hit it with
+	tcpIngressIP := getIngressPoint(&tcpService.Status.LoadBalancer.Ingress[0])
+	svcPort := tcpService.Spec.Ports[0].Port
+
+	By("creating pod to be part of service " + serviceName)
+	// TODO newRCTemplate only allows for the creation of one replica... that probably won't
+	// work so well.
+	jig.RunOrFail(f.Namespace.Name, nil)
+
+	// Hit it once before considering ourselves ready
+	By("hitting the pod through the service's LoadBalancer")
+	jig.TestReachableHTTP(tcpIngressIP, svcPort, loadBalancerLagTimeoutDefault)
+
+	sem.Ready()
+
+	if testDuringUpgrade {
+		// Continuous validation
+		wait.Until(func() {
+			By("hitting the pod through the service's LoadBalancer")
+			// TODO this is way too long of a timeout; make it shorter since we've already
+			// validated it's working.
```

Comment:
the lag timeout was specifically for cases where the LB might have changed and been subject to a long GCE operation.  If you're just testing that it is up, a shorter time is appropriate